### PR TITLE
[Impeller] Supply surface cull rect for Vulkan & GLES dispatchers

### PR DESCRIPTION
**Pull request by bdero**
Similar to Metal, have the DL cull entities early that lie entirely outside the surface texture.

### CODE REVIEW

**Review by gaaclarke — COMMENTED** [review]
LGTM, what are your plans for testing?

**Review by jonahwilliams — APPROVED** [review]
LGTM

I don't really think there are reasonable ways to test this sort of thing, besides rendering correctness and performance.

**Comment by gaaclarke** [comment]
> I don't really think there are reasonable ways to test this sort of thing, besides rendering correctness and performance.

Neither of those things would show up if this was broken here.  I plan on writing a test that uses it, we can consider that coverage at least.

**Comment by jonahwilliams** [comment]
You could stand up one of the platform view performance integration tests on Impeller/Android. I think its Skia only right now.

**Comment by jonahwilliams** [comment]
Yes that's why we have a devicelab

File: shell/gpu/gpu_surface_gl_impeller.cc [hunk 1]
```diff
@@ -104,7 +104,9 @@ std::unique_ptr<SurfaceFrame> GPUSurfaceGLImpeller::AcquireFrame(
         }
 
         impeller::DlDispatcher impeller_dispatcher;
-        display_list->Dispatch(impeller_dispatcher);
+        display_list->Dispatch(
```
Comment (jonahwilliams) [inline shell/gpu/gpu_surface_gl_impeller.cc:107]: Note: need to do the SkConversions like in the metal surface.

**Comment by flar** [comment]
Yeah, I don't see a good way to test this without instrumenting the code being tested. Does the Heisenberg Uncertainty Principle apply here?

**Comment by flar** [comment]
You can't test that the culling is happening, but you can test if the culling causes an error maybe. Draw a scene that has a mix of in-bounds, out-of-bounds, and on-the-bounds rendering ops and then make it a golden test. It would have no idea if the out-of-bounds ops got culled from that, but it would test for unintended consequences.

Could we somehow create a franken-render-target that was large but reported a small bounds and then see if the primitives were rendered outside the bounds?

**Comment by jonahwilliams** [comment]
We have benchmarks that are not currently configured to run Impeller, but could. These would show a performance improvement on this change. Benchmark improvements count as tests

**Comment by bdero** [comment]
Looking into `android_view_scroll_perf` for this at the moment.

**Pull request merged**